core/src/main/java/com/vml/es/aem/acm/core/code/CodeContext.java

```diff
@@ -0,0 +1,94 @@
+package com.vml.es.aem.acm.core.code;
+
+import com.vml.es.aem.acm.core.acl.Acl;
+import com.vml.es.aem.acm.core.code.script.ExtensionScript;
+import com.vml.es.aem.acm.core.format.Formatter;
+import com.vml.es.aem.acm.core.mock.MockContext;
+import com.vml.es.aem.acm.core.osgi.OsgiContext;
+import com.vml.es.aem.acm.core.replication.Activator;
+import com.vml.es.aem.acm.core.repo.Repo;
+import com.vml.es.aem.acm.core.script.ScriptRepository;
+import com.vml.es.aem.acm.core.script.ScriptType;
+import groovy.lang.Binding;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.slf4j.LoggerFactory;
+
+public class CodeContext {
+
+    private final OsgiContext osgiContext;
+
+    private final ResourceResolver resourceResolver;
+
+    private final Binding binding;
+
+    private final List<ExtensionScript> extensionScripts;
+
+    public CodeContext(OsgiContext osgiContext, ResourceResolver resourceResolver) {
+        this.osgiContext = osgiContext;
+        this.resourceResolver = resourceResolver;
+        this.binding = createBinding(osgiContext, resourceResolver);
+        this.extensionScripts = findExtensionScripts(resourceResolver);
+    }
+
+    private List<ExtensionScript> findExtensionScripts(ResourceResolver resourceResolver) {
+        return new ScriptRepository(resourceResolver)
+                .findAll(ScriptType.EXTENSION)
+                .map(s -> new ExtensionScript(this, s))
+                .collect(Collectors.toList());
+    }
+
+    public Binding createBinding(OsgiContext osgiContext, ResourceResolver resourceResolver) {
```

Comment:
now we have single place to define same binding for all type of scripts (including manual/auto/mocks etc)